src/devtools/client/debugger/src/actions/project-text-search.js

```diff
@@ -11,15 +11,34 @@
 
 import { isFulfilled } from "../utils/async-value";
 import { findSourceMatches } from "../workers/search";
-import { getSource, getSourceList, getSourceContent } from "../selectors";
-import { isThirdParty } from "../utils/source";
-import { loadSourceText } from "./sources/loadSourceText";
+import { getSource, getSourceContent } from "../selectors";
 import {
   statusType,
   getTextSearchOperation,
   getTextSearchStatus,
 } from "../reducers/project-text-search";
 import { ThreadFront } from "protocol/thread";
+import { groupBy } from "lodash";
+
+const mapSearchResultMatches = (matches, source) => {
+  return {
+    sourceId: source.id,
+    filepath: source?.url,
+    matches: matches.map(match => {
+      const matchStr = [...match.context]
+        .slice(match.contextStart.column, match.contextEnd.column)
+        .join("");
```

Comment:
this seems odd compared to 

```js
match.context.slice(match.contextStart.column, match.contextEnd.column)
```

Reply:
`[..."string"]` or `Array.from("string")` gives you the characters in "code points" rather than "code units". "column"s are in code points so we have to do that weird array thing.

That topic is covered in PR for the protocol https://github.com/RecordReplay/backend/pull/3499#discussion_r748615959

I also left a comment in the protocol
https://github.com/RecordReplay/backend/blob/b42ff616a14cdb7b86ea0052fe99c7041df714c0/src/control/source-worker/sourceSearch.ts#L14-L30

Reply:
It makes a different for strings with fun unicode characters